### ISSUE TITLE
`torch.audio.functional.spectrogram` does not use `normalized` parameter correctly

### ISSUE THREAD

**Issue by jojonki** [pytorch/audio#2104]
### 🐛 Describe the bug

I found `torchaudio.Spectrogram` does not use `normalized` argument correctly because the code uses hard code. Always it sets to `False`. Similar to #1007, but not griffin-lim in this case.

https://github.com/pytorch/audio/blob/main/torchaudio/functional/functional.py#L110
```python
def spectrogram(
    waveform: Tensor,
    pad: int,
    window: Tensor,
    n_fft: int,
    hop_length: int,
    win_length: int,
    power: Optional[float],
    normalized: bool,
    center: bool = True,
    pad_mode: str = "reflect",
    onesided: bool = True,
    return_complex: Optional[bool] = None,
) -> Tensor:
...
    spec_f = torch.stft(
        input=waveform,
        n_fft=n_fft,
        hop_length=hop_length,
        win_length=win_length,
        window=window,
        center=center,
        pad_mode=pad_mode,
        normalized=False, # HERE!!!
        onesided=onesided,
        return_complex=True,
    )
```

### Versions

```
Collecting environment information...
PyTorch version: 1.10.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.0.1 (arm64)
GCC version: Could not collect
Clang version: 13.0.0 (clang-1300.0.29.3)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.7 (default, Nov 14 2021, 14:30:16)  [Clang 13.0.0 (clang-1300.0.29.3)] (64-bit runtime)
Python platform: macOS-12.0.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.10.1
[pip3] torchaudio==0.10.1
[conda] Could not collect
```

**Comment by mthrok** [comment]
Hi @jojonki 

`normalized` parameter is not passed to stft function, but used in couple of lines later.

https://github.com/pytorch/audio/blob/9397cd5b9c4389baa8b3891b3ba12aa2ae07100f/torchaudio/functional/functional.py#L117-L119

Is this not serving your purpose?

**Comment by jojonki** [comment]
Hi @mthrok . Thank you! I missed the point.

Why does the code normalize the spectrogram after `stft` function while `stft` function has `normalized` argument.
Are there any differences?

**Comment by SergeantChris** [comment]
Why is the normalization performed with the `window` (e.g. `torch.hann_window` in the default) instead of the `waveform`? 
The normalization currently implemented is actually exactly the same regardless of the input signal.

Shouldn't it be `spec_f /= waveform.pow(2.0).sum().sqrt()` as we want to take into account the energy of the signal and not the window function?

**Comment by skim0514** [comment]
Resolved in #2554
